### PR TITLE
XMC4XXX uart poll_in fixes

### PR DESCRIPTION
**Pull request by talih0**

### CODE REVIEW

**Comment by dcpleung** [comment]
You need to add a body to the commit message to the third commit.

**Comment by dcpleung** [comment]
Need to fix the compliance check failure.

File: drivers/serial/uart_xmc4xxx.c [hunk 1]
```diff
@@ -23,7 +23,11 @@ static int uart_xmc4xxx_poll_in(const struct device *dev, unsigned char *c)
 {
 	const struct uart_xmc4xx_config *config = dev->config;
 
-	*(uint16_t *)c = XMC_UART_CH_GetReceivedData(config->uart);
+	if (!XMC_USIC_CH_GetReceiveBufferStatus((XMC_USIC_CH_t *)config->uart)) {
+		return -1;
+	}
+
+	*c = XMC_UART_CH_GetReceivedData((XMC_USIC_CH_t *)config->uart);
```
[the remark this answers is not in the document]
Comment (talih0) [reply drivers/serial/uart_xmc4xxx.c:30]: Yep, thanks, good point.

[the remark this answers is not in the document]
Comment (talih0) [reply drivers/serial/uart_xmc4xxx.c:30]: The compiler will do an implicit cast from uint16_t to unsigned char and truncate the value (there are no warnings when compiling). I personally don't think it's needed. But fine to add it if you prefer the other style. Let me know :)

**Review by dcpleung — DISMISSED** [review]

File: drivers/serial/uart_xmc4xxx.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (npal-cy) [reply drivers/serial/uart_xmc4xxx.c]: yes compilers know how to deal this this...But this place will show warning in different code static analyzer with MISRA checker.

[the remark this answers is not in the document]
Comment (talih0) [reply drivers/serial/uart_xmc4xxx.c]: Ok, I've updated the patch.

**Review by dcpleung — APPROVED** [review]

**Review by parthitce — APPROVED** [review]
Thanks for fixing it. LGTM

**Pull request merged**